### PR TITLE
Add speech modifier to zombie tongue

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

A zombie rotten tongue has a complex language modifier.

#### The language modifier works by:

1. All occurrences of characters "eiou" (case-insensitive) are replaced with "r". 
2. All characters other than "zhrgbmna .!?-" (case-insensitive) are stripped.
3. Multiple spaces are replaced with a single.
4. Lower-case "r" at the end of words replaced with "rh".
5. An "a" or "A" by itself will be replaced with "hra".
6. The first character is capitalised.

#### Some interesting dialogue examples:

> _Bab, am  gaa habbah abah  zah namrh ah Bh!rh!b?_
> _Bob, are you happy  about the death of Philip?_

> _Zah bang bang man ganna harm mah zambah?_
> _Will the Zombie Hunter attack me?_

>  _Mah zambah nah harm brazzarz.
>  I do not hurt brothers._

>  _Mah zambah ganna gangbang harmanz zammarrar.
>  I will kill humans tomorrow._

>  _Mah zambah am nah habbah, an mah zambah gab, -Graaaagh!-
> I am not happy, and I say "Graaaagh!"_

---

The language idea was taken from a zombie game back in 2005 called [Urban Dead](https://en.wikipedia.org/wiki/Urban_Dead#:~:text=Urban%20Dead%20was%20created%20by,play%2Dby%2Dmail%20games.).  It's no longer developed and I made all the code myself while following the given language rule structures. 

[Zombie Speech Translator](http://wiki.urbandead.com/index.php/Zombie_Speech_Translators#How_Urban_Dead_processes_Zombie_speech)
[Zombie Language Examples](http://wiki.urbandead.com/index.php/Zamgrh)
[Zombie Dictionary](http://wiki.urbandead.com/index.php/Guides:The_Zombie_Lexicon#Pre_Death_Rattle_Zombie_Expressions)

#### Abracadabra - The Steve Miller Band

> _Ah raab zha brahnz ahn zarh hagh,_ (I love the brains in your head)
> _Ah ganna barg abgrah gangbang,_ (I'm gonna eat them when you're dead)
> _Az rahnah zarh ranz ahn hahg ahahz,_ (Now as you run and hide away)
> _Zarh harh mah zambah az hah zahz:_ (You hear my zombie as he says:)
> _Abra-abra-gababra,_ (Abra-abra-cadabra)
> _Ah ganna rahg arg ahn grab zarh!_ (I'm gonna reach out and grab ya!)
> _Abra-abra-gababra,_ (Abra-abra-cadabra)
> _Ah ganna rahg arg ahn grab zarh!_ (I'm gonna reach out and grab ya!)

## Changelog

:cl:
add: Rotten zombie tongue has a new speech modifier that converts spoken language into zombie sentences. If the person speaking is a high-functioning zombie this is bypassed.
/:cl:

### CODE REVIEW

**Comment by dragomagol** [comment]
I like that zombies get their own language and the way you've set it up! Did you get maint approval for this species change to skeletons and zombies?

I'm not sure I agree with zombies understanding calcic and skeletons understanding zombie. Don't feel especially strongly about this.

I disagree more strongly with the removal of the aphasia language for this (mostly because it's zombie-themed, with `I` becoming `Mah zombah`) where aphasia is just a general inability to communicate properly).

**Comment by itseasytosee** [comment]
> I like that zombies get their own language and the way you've set it up! Did you get maint approval for this species change to skeletons and zombies?
> 
> I'm not sure I agree with zombies understanding calcic. Don't feel especially strongly about this.
> 
> I disagree more strongly with the removal of the aphasia language for this, mostly because it's zombie-themed, with `I` becoming `Mah zambah`, where aphasia is just a general inability to communicate properly).

Is there any other place aphasia is used?

**Comment by dragomagol** [comment]
> > I like that zombies get their own language and the way you've set it up! Did you get maint approval for this species change to skeletons and zombies?
> > I'm not sure I agree with zombies understanding calcic. Don't feel especially strongly about this.
> > I disagree more strongly with the removal of the aphasia language for this, mostly because it's zombie-themed, with `I` becoming `Mah zambah`, where aphasia is just a general inability to communicate properly).
> 
> Is there any other place aphasia is used?

Brain damage. I'm not sure what you mean by "other" if you mean something besides that?

**Comment by itseasytosee** [comment]
> > > I like that zombies get their own language and the way you've set it up! Did you get maint approval for this species change to skeletons and zombies?
> > > I'm not sure I agree with zombies understanding calcic. Don't feel especially strongly about this.
> > > I disagree more strongly with the removal of the aphasia language for this, mostly because it's zombie-themed, with `I` becoming `Mah zambah`, where aphasia is just a general inability to communicate properly).
> > 
> > 
> > Is there any other place aphasia is used?
> 
> [Brain damage](https://tgstation13.org/wiki/Guide_to_Traumas#Severe). I'm not sure what you mean by "other" if you mean something besides that?

Yeah, it is weird that brain damaged people would start talking like zombies

**Comment by timothymtorres** [comment]
> I like that zombies get their own language and the way you've set it up! Did you get maint approval for this species change to skeletons and zombies?

Not yet.  I was under the impression that the species rule applied to the base starting species, if that isn't the case then I'll be sure to double check with maintainers next time.

> I'm not sure I agree with zombies understanding calcic. Don't feel especially strongly about this.

So zombies and skeletons are apart of the Halloween species.  They are brethren and while not being able to speak each others language, it would make sense that they can understand each other.  Y'know being undead and all.

> I disagree more strongly with the removal of the aphasia language for this, mostly because it's zombie-themed, with `I` becoming `Mah zambah`, where aphasia is just a general inability to communicate properly).

The aphasia language was heavily underutilized.  It can only be gained by one source in the entire game:

https://github.com/tgstation/tgstation/blob/8016f3576ee5fb7b286c3b3f4e6783bb8a02c06e/code/datums/brain_damage/severe.dm#L23-L38

The language sounds were also pretty bland compared to other languages that stand out.  I felt the description about it being a language for brain damaged mobs suited zombies perfectly since they lack high intelligence. (plus aphasia uses the brain icon, which is PERFECT for zombie language)

Fear not though, there is still a different source of aphasia:

https://github.com/tgstation/tgstation/blob/2ff790863981eda4e2cf1fe2dcdc65040497f0b2/code/datums/brain_damage/mild.dm#L184-L226

Although this is not a language itself but a modifier. (like french or chav language modifiers)

**Comment by dragomagol** [comment]
> The aphasia language was heavily underutilized.  It can only be gained by one source in the entire game:
> 
> https://github.com/tgstation/tgstation/blob/8016f3576ee5fb7b286c3b3f4e6783bb8a02c06e/code/datums/brain_damage/severe.dm#L23-L38

And this makes perfect sense, because it's an inability to communicate.

> The language sounds were also pretty bland compared to other languages that stand out.  I felt the description about it being a language for brain damaged mobs suited zombies perfectly since they lack high intelligence. (plus aphasia uses the brain icon, which is PERFECT for zombie language)
> 
> Fear not though, there is still a different source of aphasia:
> 
> https://github.com/tgstation/tgstation/blob/2ff790863981eda4e2cf1fe2dcdc65040497f0b2/code/datums/brain_damage/mild.dm#L184-L226
> 
> Although this is not a language itself but a modifier. (like french or chav language modifiers)

You still haven't addressed my primary criticism of people with aphasia sounding like zombies.

**Comment by dragomagol** [comment]
The DNM here is for the removal of aphasia. 

If you want to give aphasia a new icon and use the brain for zombies that's fine. Or you could make the zombie one be a gravestone/something else.

File: strings/zombie_replacement.json [hunk 1]
```diff
@@ -1,163 +1,255 @@
...
+	  "zzzz": ["chill", "relax", "sleep", "calm", "peace", "dream"],
+	  "bah": ["bad", "evil", "but"],
+	  "gag": ["gross", "nasty", "vomit", "smelly", "stinky", "stench", "foul", "quiet", "silence", "shutup"],
+	  "gammah": ["give", "gimmie"],
+	  "gah": ["get", "go"],
+	  "gargazz": ["body", "corpse"],
+	  "rag": ["uniform", "suit", "rag", "clothing", "jumpsuit"],
+	  "garbagz": ["garbage", "janitor", "custodian", "trash", "clean", "dirt", "filth", "mess", "junk"],
+	  "bagman": ["doctor", "medic", "healer", "paramedic", "curator", "librarian"],
+	  "gangbang": ["death", "kill", "die", "slay", "ERP", "punish", "detain", "prison", "prisoner", "brig", "cell", "jail", "perma", "permabrig"],
```
Comment (itseasytosee) [inline strings/zombie_replacement.json:189]: 🤨

Comment (timothymtorres) [reply strings/zombie_replacement.json:189]: chrome_ZJD88ZsZ6S

**Comment by timothymtorres** [comment]
I tossed out the zombie language and just moved everything into the tongue speech modifier.  Much simpler and easier that way.

File: code/modules/surgery/organs/tongue.dm [hunk 2]
```diff
@@ -286,21 +286,68 @@
...
 
-		if(!(copytext(inserttext, -3) == "..."))//3 == length("...")
-			message_list[insertpos] = inserttext + "..."
+/obj/item/organ/internal/tongue/zombie/modify_speech(datum/source, list/speech_args)
+	var/mob/living/carbon/human/user = source
 
-		if(prob(20) && message_list.len > 3)
-			message_list.Insert(insertpos, "[pick("BRAINS", "Brains", "Braaaiinnnsss", "BRAAAIIINNSSS")]...")
+	// only high functioning zombies can bypass speech limitations
+	if(iszombie(user) && !is_species(user, /datum/species/zombie/infectious))
```
Comment (JohnFulpWillard) [inline code/modules/surgery/organs/tongue.dm:312]: I think all zombies should speak like this, or anyone with this tongue, it's more comedic.

Comment (timothymtorres) [reply code/modules/surgery/organs/tongue.dm]: Applied with https://github.com/tgstation/tgstation/pull/69899/commits/5288fbc7fcf1350fd62665389b1dabf2442dc2e2

**Pull request merged**